### ISSUE TITLE
7.49.1 needlessly breaks header parsers' expectations

### ISSUE THREAD

**Issue by m6w6** [curl/curl#888]
7.49.1 apparently changed the string "HTTP/2.0" to "HTTP/2", which will be passed to the header callback.

We all know that the protocol is called HTTP/2 and not HTTP/2.0, but this change needlessly breaks header parsers' expectations for little to no gain in x.y.Z release.

Even the spec uses "HTTP/2.0" in the connection preface:
https://tools.ietf.org/html/rfc7540#section-3.5

**Comment by bagder** [comment]
I consider the "HTTP/2.0" in the past to be a bug as the protocol isn't named like that. We added the string back in the days where it still wasn't decided that the ".0" was to get dropped and then we never went back and fixed it (we had a few other traces of early HTTP/2 implementation artifacts that we've fixed over time).

I also wanted internal consistency of what we call the protocol as I think it makes sense for us to refer to the protocol with a single name and style all through the project and libcurl in particular.

Others also already use "HTTP/2" to refer to the protocol, for example in Apache logs.

With the introduction of `CURLINFO_HTTP_VERSION` (coming in the pending 7.50.0 release), the need to parse the HTTP protocol version by applications should hopefully be somewhat lesser than before.

What to do now? Well, we already have the 7.49.1 release out in the wild that provides this output. Should we bite the bullet and say that we fixed the bug this way and make others adapt or should we revert that decision and say that it should say "2.0" forever onward for compatibility's sake?

I'd be interested if others have more feedback or additional viewpoints on this. I'll listen.
